### ISSUE TITLE
Support stencilOpSeparate & stencilFuncSeparate in webgl2 ?

### ISSUE THREAD

**Issue by shawn0326** [mrdoob/three.js#20120]
Now the properties of Material only support `gl.stencilOp` & `gl.stencilFunc`. 
Can we add some properties to Material like `stencilZFailBack` to support separated stencil settings in webgl2 ?

**Comment by gkjohnson** [comment]
The `separate` version of the stencil APIs is actually [available in WebGL1, as well](https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/stencilFuncSeparate). In some cases it would cut down on stenciled draw calls by half when the operations for front and back faces need to be different but everything can still be achieved with the current API. 

Just to think through the API a bit are you imagining a `Back` variant for all 7 material stencil fields? Would anything enable the user to specify that both front and back faces can share stencil settings? Or would you always have to set both front and back fields separately and incur double the amount of gl stencil set function calls?

**Comment by shawn0326** [comment]
> The separate version of the stencil APIs is actually available in WebGL1, as well.

yes, but i noticed that the Status of these Specification are Recommendation in webgl1. some browsers may have problom.

> Just to think through the API a bit are you imagining a `Back` variant for all 7 material stencil fields? 

in order to do this, i think these 6 properties should have `Back` variant:
`stencilFunc`  `stencilRef`  `stencilFuncMask` `stencilFail` `stencilZFail`  `stencilZPass`
like
`stencilFuncBack`  `stencilRefBack`  `stencilFuncMaskBack` `stencilFailBack` `stencilZFailBack`  `stencilZPassBack`

> Would anything enable the user to specify that both front and back faces can share stencil settings? Or would you always have to set both front and back fields separately and incur double the amount of gl stencil set function calls?

My idea: the default value of these properties is null. ( like texture.internalFormat ). if these properties are set explicitly, use stencilOpSeparate or stencilFuncSeparate. otherwise just use stencilOp & stencilFunc.

**Comment by shawn0326** [comment]
BTW, i tested gl.stencilOpSeparate & gl.stencilFuncSeparate in lastest chrome and firefox, works fine~
